b2/thread_pool.py

```diff
@@ -0,0 +1,112 @@
+######################################################################
+#
+# File: b2/thread_pool.py
+#
+# Copyright 2016 Backblaze Inc. All Rights Reserved.
+#
+# License https://www.backblaze.com/using_b2_code.html
+#
+######################################################################
+
+from __future__ import print_function
+
+import threading
+from six.moves import range
+from six.moves.queue import Queue
+
+# this is added to the task queue to tell the threads to stop
+SHUT_DOWN_TOKEN = 'shut_down'
```

Comment:
I suggest to change it to None or an object - comparing strings via `is` operator is not recommended